### PR TITLE
Various code cleanups

### PR DESCRIPTION
**Pull request by martinpitt**
I tested them locally a little, but I will trigger a bunch of representative project tests against it.  In particular, cockpit's check-bots-api covers image-customize rather thoroughly.

### CODE REVIEW

File: image-customize [hunk 1]
```diff
@@ -144,6 +116,39 @@ def install_packages(machine_instance, packagelist, install_command):
         machine_instance.execute(install_command + " " + ' '.join(allpackages), timeout=1800)
 
 
+#
```
Comment (allisonkarlitskaya) [inline image-customize:118]: I rather enjoy defining a `main()` function and adding at the very end:

```
if __name__ == '__main__':
    main()
```

Comment (martinpitt) [reply image-customize:118]: Sure! I'll let the first wave of tests run, and then change this.

File: image-customize [hunk 2]
```diff
@@ -143,30 +143,33 @@ parser.add_argument('-n', '--no-network', action='store_true', help='Do not conn
 parser.add_argument('image', help='The image to use (destination name when using --base-image)')
 args = parser.parse_args()
 
+if not (args.run_command or args.install or args.script or args.upload or args.resize):
```
Comment (allisonkarlitskaya) [inline image-customize:141]: For what it's worth, I've never liked how this works.  I'd sort of prefer if we built a single array and executed the operations in the order they were given, like how `vm-run` does with `-s` and `-e`.  In the case of `vm-run` the implementation isn't beautiful, but it's conceptually better.

**Review by allisonkarlitskaya — DISMISSED** [review]
Everything here looks good, and are nice cleanups (thanks!) but there are a couple of suggestions for even more changes.

File: image-customize (hunk 2 above)
Comment (martinpitt) [reply image-customize:141]: Right, I can't say I like it much either. There are good reasons to run a command before and after installing packages, and it's totally not obvious why --run-command runs before, but --script runs after. (It usually makes sense in practice, but being explicit is better). argparse does not make this easy, but I'll add it to my todo list to experiment with that.

File: image-customize (hunk 1 above)
Comment (martinpitt) [reply image-customize]: Done. This spotted another little issue.

**Review by allisonkarlitskaya — APPROVED** [review]
approved, subject to you promising to unify the list :)

**Comment by allisonkarlitskaya** [comment]
> approved, subject to you promising to unify the list :)

→ #2860

**Pull request merged**